### PR TITLE
Add tests for RubygemSearchable

### PR DESCRIPTION
**Pull request by sonalkr132**

### CODE REVIEW

File: test/unit/rubygem_searchable_test.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    setup do
+      example_gem1 = create(:rubygem, name: "example_gem1")
+      example_gem2 = create(:rubygem, name: "example_gem2")
+      create(:version, rubygem: example_gem1, indexed: false)
+      create(:version, rubygem: example_gem2)
+      import_and_refresh
+    end
+
+    should "filter yanked gems from the result" do
+      response = Rubygem.elastic_search "example_gem2"
```
Comment (dwradcliffe) [inline test/unit/rubygem_searchable_test.rb]: This only searches for the indexed gem name so it isn't testing the filter.

File: app/models/concerns/rubygem_searchable.rb [hunk 2]
```diff
@@ -11,12 +11,11 @@ class SearchDownError < StandardError; end
...
     delegate :index_document, to: :__elasticsearch__
     delegate :update_document, to: :__elasticsearch__
-    delegate :delete_document, to: :__elasticsearch__
 
-    # These are not used, because we trigger the ES index from the Pusher class
+    # These are not used, because we create index in the Pusher class
     # after_commit -> { delay.index_document  }, on: :create
+    # we update index when gem is yanked and when we bulk update downloads
     # after_commit -> { delay.update_document }, on: :update
-    # after_commit -> { delay.delete_document }, on: :destroy
```
Comment (dwradcliffe) [inline app/models/concerns/rubygem_searchable.rb]: Let's just remove all this commented stuff

**Comment by sonalkr132** [comment]
updated :+1:

**Comment by dwradcliffe** [comment]
Great start here! I'm going to merge this now. I have another case I'd like to test but we can add that later.

**Comment by arthurnn** [comment]
❤️

**Pull request merged**